src/Microsoft.ML.Data/DataDebuggerPreview.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.ML.Data
+{
+    /// <summary>
+    /// This class represents an eager 'preview' of a <see cref="IDataView"/>.
+    /// </summary>
+    public sealed class DataDebuggerPreview
+    {
+        internal static class Defaults
+        {
+            public const int MaxRows = 100;
+        }
+
+        public Schema Schema { get; }
+        public ColumnInfo[] ColumnView { get; }
+        public RowInfo[] RowView { get; }
```

Comment:
>RowInfo[] RowView [](start = 15, length = 17)

Now that we have it, what do you think about shifting this sort of thing to start using `ImmutableArray<ColumnInfo>` and `ImmutableArray<RowInfo>`?